llvm/lib/Transforms/IPO/AlwaysInliner.cpp

```diff
@@ -38,6 +38,7 @@ bool AlwaysInlineImpl(
   SmallSetVector<CallBase *, 16> Calls;
   bool Changed = false;
   SmallVector<Function *, 16> InlinedFunctions;
+  SmallVector<Function *, 16> WorkList;
   for (Function &F : M) {
```

Comment:
I think we can do `for (Function &F: make_early_inc_range(M))` to enable deleting in this loop

Reply:
yes, thanks!